src/geohash_helper.c

```diff
@@ -100,34 +100,41 @@ uint8_t geohashEstimateStepsByRadius(double range_meters, double lat) {
  * Since this function is currently only used as an optimization, the
  * optimization is not used for very big radiuses, however the function
  * should be fixed. */
-int geohashBoundingBox(double longitude, double latitude, double radius_meters,
-                       double *bounds) {
+int geohashBoundingBox(GeoShape *shape, double *bounds) {
     if (!bounds) return 0;
-
-    bounds[0] = longitude - rad_deg(radius_meters/EARTH_RADIUS_IN_METERS/cos(deg_rad(latitude)));
-    bounds[2] = longitude + rad_deg(radius_meters/EARTH_RADIUS_IN_METERS/cos(deg_rad(latitude)));
-    bounds[1] = latitude - rad_deg(radius_meters/EARTH_RADIUS_IN_METERS);
-    bounds[3] = latitude + rad_deg(radius_meters/EARTH_RADIUS_IN_METERS);
+    double longitude = shape->xy[0];
+    double latitude = shape->xy[1];
+    double height = shape->conversion * (shape->type == CIRCULAR_TYPE ? shape->radius : shape->height);
+    double width = shape->conversion * (shape->type == CIRCULAR_TYPE ? shape->radius : shape->width);
+
+    bounds[0] = longitude - rad_deg(height/EARTH_RADIUS_IN_METERS/cos(deg_rad(latitude)));
+    bounds[2] = longitude + rad_deg(height/EARTH_RADIUS_IN_METERS/cos(deg_rad(latitude)));
+    bounds[1] = latitude - rad_deg(width/EARTH_RADIUS_IN_METERS);
+    bounds[3] = latitude + rad_deg(width/EARTH_RADIUS_IN_METERS);
```

Comment:
@yangbodong22011 isn't latitude related to height and longitude to width, as per: 
![image](https://user-images.githubusercontent.com/5832149/100267388-4153dd80-2f4b-11eb-9e7a-9d8c7008740f.png)

-----
On a second comment:
WDYT about fixing ( if you agree on the lat/long discussion above ) and replacing:
```
bounds[0] = longitude - rad_deg(height/EARTH_RADIUS_IN_METERS/cos(deg_rad(latitude)));
bounds[2] = longitude + rad_deg(height/EARTH_RADIUS_IN_METERS/cos(deg_rad(latitude)));
bounds[1] = latitude - rad_deg(width/EARTH_RADIUS_IN_METERS);
bounds[3] = latitude + rad_deg(width/EARTH_RADIUS_IN_METERS);
```

by:
```
const double long_delta = rad_deg(width/EARTH_RADIUS_IN_METERS/cos(deg_rad(latitude)));
const double lat_delta = rad_deg(height/EARTH_RADIUS_IN_METERS);
bounds[0] = longitude - long_delta;
bounds[2] = longitude + long_delta;
bounds[1] = latitude - lat_delta;
bounds[3] = latitude + lat_delta;
```
It will cut the computation by half on that function. PS (as a bonus): can we also ensure constant correcteness of  longitude,latitude,height, and width.

Reply:
@filipecosta90 I agree with you, thanks.